empress/support_files/js/summary-helper.js

```diff
@@ -0,0 +1,65 @@
+define([], function() {
+    /**
+     * @class SummaryHelper
+     * Class functions for generating summary used in empress
+     */
+    function SummaryHelper() {}
+
+    /**
+     * Returns unifrac value for 2 set of sample IDs given biom table
+     * @param {BiomTable} biomTable - biom table that contains info about samples
+     * @param {BPTree} tree - bp tree
+     * @param {Array} sIds1 - First array of sample IDs to calculate unifrac for
+     * @param {Array} sIds2 - Second array of sample IDs to calculate unifrac for
+     *
+     * @return{Number}
+     */
+    SummaryHelper.unifrac = function(biomTable, tree, sIds1, sIds2) {
+        var uniq = 0;
+        var total = 0;
+        var uniqObs1 = biomTable.getObjservationUnionForSamples(sIds1);
+        var uniqObs2 = biomTable.getObjservationUnionForSamples(sIds2);
+
+        // Elements are in postorder with first element at index 0
+        // To keep track of if the node is in the sample1
+        var count1 = new Uint8Array(tree.names_.length);
+        // To keep track of if the node is in the sample2
+        var count2 = new Uint8Array(tree.names_.length);
+
+        // Based on the info in count1 and count2, calculate branch length
+        // Some optimization can be done, maybe perform the operation on the arrays
+        for (var i = 1; i <= tree.size; i++) {
+            if (tree.postorderselect(i) !== tree.root()) {

```

Comment:
```suggestion
for (var i = 1; i < tree.size; i++) {
```

tree.root() will always be the last node in postorder so no need for the if statement